pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3201,7 +3239,20 @@ func (c *Cloud) getInstancesByNodeNamesCached(nodeNames sets.String) ([]*ec2.Ins
 			return c.lastInstancesByNodeNames, nil
 		}
 	}
-	names := aws.StringSlice(nodeNames.List())
+	instances, err := c.getInstancesByNodeNames(nodeNames.List())
+
+	if err != nil {
+		return nil, err
```

Comment:
Do you mean `getInstancesByNodeNames` returns `nil,nil` when no instances are found?